crates/smelt-graph/src/executor/docker.rs

```diff
@@ -66,38 +78,46 @@ impl Executor for DockerExecutor {
         let root = global_data.get_smelt_root();
         let root_as_str = root
             .to_str()
-            .expect("Smelt root couldnt be converted to string ")
-            .to_string();
+            .expect("Smelt root couldnt be converted to string ");
 
         let command_default_dir = command.working_dir.clone();
         let silent = global_data.get_smelt_cfg().silent;
 
         // "Prepares" the workspace for this command -- creates a directory at path
         // {SMELT_ROOT}/smelt-out/{COMMAND_NAME}
-        let Workspace {
-            script_file,
-            mut stdout,
-        } = prepare_workspace(&command, root.clone(), command_default_dir.as_path()).await?;
-
-        // The "default" bind mount for all commands is smelt root -- in expectation, this should
-        // mount the git root in to the container, at the same path as it has on the host
-        // filesystem
-        let base_binds = vec![format!("{}:{}", root_as_str, root_as_str)];
-        let binds = self
-            .additional_mounts
-            .iter()
-            .fold(base_binds, |mut val, b| {
-                val.push(format!("{}:{}", b.0, b.1));
-
-                val
-            });
-
-        let cmd = vec![shell.to_string(), script_file.to_str().unwrap().to_string()];
+        let mut stdout = File::open("/dev/null").await?;
+
+        let cmd = match self.docker_run_mode {
+            DockerRunMode::Local => {
+                let workspace = prepare_workspace(&command, root.clone(), command_default_dir.as_path()).await?;
+                stdout = workspace.stdout;
+                vec![shell.to_string(), workspace.script_file.to_str().unwrap().to_string()]
+            }
+            DockerRunMode::Remote => {
+                vec!["bash".to_string(), "-c".to_string(), command.script.clone().join(" && ")]
```

Comment:
the assumption that some commands make is that `smelt-out/${COMMAND_NAME}` is a path that exists  -- `prepare_workspace` is responsible for this  for this directory creation